src/runtime/runtime.cs

```diff
@@ -1193,7 +1193,11 @@ internal static bool PyString_Check(IntPtr ob)
 
         internal static IntPtr PyString_FromString(string value)
         {
+#if PYTHON3
+            return PyUnicode_FromKindAndData(_UCS, value, value.Length);
+#elif PYTHON2
             return PyString_FromStringAndSize(value, value.Length);
```

Comment:
Maybe check here whether the string is ASCII? I'm not sure where this one is used, but we should probably distinguish the case where we actually want Unicode (and should also return that on Python 2) from the cases that should be ASCII-only on Python 2 (like identifiers).

Reply:
@filmor can you file a separate issue for this with a suggested fix? this PR is now ready to merge!